cmd/autoowners/main.go

```diff
@@ -196,74 +238,64 @@ func writeYAML(path string, data interface{}, prefix []string) (rerr error) {
+		orgRepo.SimpleConfig.Reviewers = resolveAlias(orgRepo.SimpleConfig.Reviewers, orgRepo.RepoAliases)
+		orgRepo.SimpleConfig.RequiredReviewers = resolveAlias(orgRepo.SimpleConfig.RequiredReviewers, orgRepo.RepoAliases)
+	} else {
+		for k, v := range orgRepo.FullConfig.Filters {
+			orgRepo.FullConfig.Filters[k] = repoowners.Config{
+				Approvers:         resolveAlias(v.Approvers, orgRepo.RepoAliases),
+				Reviewers:         resolveAlias(v.Reviewers, orgRepo.RepoAliases),
+				RequiredReviewers: resolveAlias(v.RequiredReviewers, orgRepo.RepoAliases),
+				Labels:            v.Labels,
+			}
 		}
 	}
-	return owners
 }
 
-// resolveOwnerAliases checks whether the orgRepo includes any
-// owner aliases, and attempts to resolve them to the appropriate
-// set of owners.  Returns an owners which replaces any
-// matching aliases with the set of owner names belonging to that alias.
-func (orgRepo *orgRepo) resolveOwnerAliases() *owners {
-	if orgRepo.Aliases == nil || len(orgRepo.Aliases.Aliases) == 0 {
-		return orgRepo.Owners
+func resolveAlias(names []string, alias RepoAliases) []string {
+	if names == nil {
+		return nil
 	}
-
-	return &owners{
-		Approvers:         resolveAliases(orgRepo.Aliases, orgRepo.Owners.Approvers),
-		Reviewers:         resolveAliases(orgRepo.Aliases, orgRepo.Owners.Reviewers),
-		RequiredReviewers: orgRepo.Owners.RequiredReviewers,
-		Labels:            orgRepo.Owners.Labels,
+	result := sets.NewString()
+	for _, name := range names {
+		login := github.NormLogin(name)
+		if val, ok := alias[login]; ok {
+			expand := val.List()
+			logrus.WithField("alias", login).WithField("expand", expand).Debug("resolving alias ...")
+			result.Insert(expand...)
+		} else {
+			result.Insert(login)
+		}
 	}
+	return result.List()
 }
 
 func (orgRepo *orgRepo) writeOwners(whitelist []string) (err error) {
 	for _, directory := range orgRepo.Directories {
 		if inWhitelist(directory, whitelist) {
-			logrus.WithField("directory", directory).Info("Ignoring the directory in the white list.")
+			logrus.WithField("directory", directory).Info("Ignore the directory in the white list.")
 			continue
 		}
 		path := filepath.Join(directory, "OWNERS")
 		err := os.Remove(path)
 		if err != nil && !os.IsNotExist(err) {
 			return err
 		}
-		if orgRepo.Owners == nil {
-			continue
+		orgRepo.resolveOwnerAliases()
+		var content interface{}
+		content = orgRepo.SimpleConfig
+		if orgRepo.SimpleConfig.Empty() {
+			content = orgRepo.FullConfig
 		}
-
-		err = writeYAML(path, orgRepo.resolveOwnerAliases(), []string{
+		logrus.WithField("path", path).Debug("Writing to path ...")
+		err = writeYAML(path, content, []string{
 			doNotEdit,
 			fmt.Sprintf(
-				"# from https://github.com/%s/%s/blob/%s/OWNERS\n",
+				//prow github client does not parse out the latest sha of the file
+				"# from https://github.com/%s/%s/blob/master/OWNERS when master is at %s\n",
```

Comment:
nit: you can link the OWNERS file at that commit with 

```go
fmt.Sprintf("https://github.com/%s/%s/blob/%s/OWNERS", orgRepo.Organization, orgRepo.Repository, orgRepo.Commit)
```

Reply:
Tried.
the commit is for HEAD ... if the file is not changed in that commit, then that link is broken.

Actually, when we get file content, github returns us the latest commit on that file, but prow's github client does not return the SHA of the commit.

Reply:
Wait ... am talking crazy ... retesting ...

Reply:
Fixed.